browser/lib/transport/jsonptransport.js

```diff
@@ -17,8 +18,10 @@ var JSONPTransport = (function() {
 	}
 	Utils.inherits(JSONPTransport, CometTransport);
 
-	JSONPTransport.isAvailable = function() { return true; };
-	ConnectionManager.httpTransports['jsonp'] = ConnectionManager.transports['jsonp'] = JSONPTransport;
+	JSONPTransport.isAvailable = function() { return isSupported; };
+	if(isSupported) {
```

Comment:
If we do not make this change (ie we still add the `jsonp` member) then if anyone attempts to request jsonp they will get an error saying it isn't supported (which I think is preferable), instead of saying it's unrecognised.